resource/config_map.go

```diff
@@ -0,0 +1,201 @@
+
+		desired, err = c.desiredFn(ctx, obj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		validateDesiredObject(desired, c.namespace, c.name)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found desired ConfigMap %#q", printName))
+	}
+
+	var current *corev1.ConfigMap
+	{
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding current ConfigMap %#q", printName))
+
+		current, err = c.k8sClient.CoreV1().ConfigMaps(c.namespace).Get(c.name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find current ConfigMap %#q", printName))
+
+		} else if err != nil {
+			return microerror.Mask(err)
+
+		} else {
+			c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found current ConfigMap %#q", printName))
+		}
+	}
+
+	if current != nil {
+		c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("updating ConfigMap %#q", printName))
+
+		toUpdate := newConfigMapToUpdate(current, desired)
+
+		if toUpdate != nil {
+			_, err := c.k8sClient.CoreV1().ConfigMaps(c.namespace).Update(toUpdate)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("updated ConfigMap %#q", printName))
+
+		} else {
+			c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ConfigMap %#q is up to date", printName))
+		}
+
+	} else {
+		c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("creating ConfigMap %#q", printName))
+
+		_, err := c.k8sClient.CoreV1().ConfigMaps(c.namespace).Create(desired)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("created ConfigMap %#q", printName))
+	}
+
+	return nil
+}
+
+func (c *ConfigMap) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	var err error
+
+	printName := newPrintName(c.namespace, c.name)
+
+	{
+		c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("deleting ConfigMap %#q", printName))
+
+		err = c.k8sClient.CoreV1().ConfigMaps(c.namespace).Delete(c.name, &metav1.DeleteOptions{})
+		if apierrors.IsNotFound(err) {
+			c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("ConfigMap %#q already deleted", printName))
+
+		} else if err != nil {
+			return microerror.Mask(err)
+
+		} else {
+			c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("deleted ConfigMap %#q", printName))
+		}
+	}
+
+	return nil
+}
+
+// newConfigMapToUpdate creates a new ConfigMap to update based on current and
+// desired one. If there is no update nil is returned.
+func newConfigMapToUpdate(current, desired *corev1.ConfigMap) *corev1.ConfigMap {
+	merged := current.DeepCopy()
+
+	if !reflect.DeepEqual(current.Annotations, desired.Annotations) {
+		merged.Labels = desired.Annotations
```

Comment:
Is this still correct? merged.**Labels** = desired.**Annotations**?

Reply:
Still wrong :D